#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_ws_addarg_retval_fix.dpatch by  <jblache@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix a bad return value in ws_addarg() when updating a header value,
## DP: leading the caller to believe an out of memory condition happened.
## DP: Patch by Jeremy Kerr <jk@ozlabs.org>

@DPATCH@
diff -urNad mt-daapd-0.9~r1696~/src/webserver.c mt-daapd-0.9~r1696/src/webserver.c
--- mt-daapd-0.9~r1696~/src/webserver.c	2008-09-12 18:44:16.000000000 +0200
+++ mt-daapd-0.9~r1696/src/webserver.c	2008-09-12 18:45:25.537436627 +0200
@@ -1514,7 +1514,7 @@
             free(newkey);
             free(pnew);
             WS_EXIT();
-            return 0;
+            return TRUE;
         }
         current=current->next;
     }
